### PR TITLE
Support Export of Openai Whisper [Batched decoding ver]

### PR DESCRIPTION
**Pull request by shubhambhokare1**
Depending on merge of https://github.com/microsoft/onnxruntime/pull/17316

Two additional inputs added to the encoderdecoderinit subgraph (for the first decoder run):

- left_pad_mask: left pad mask is added to qk node in the qkv attention function
- position_ids: used to select indices of positional embeddings [```x = self.token_embedding(x) + self.positional_embedding[position_ids, :]```]

### CODE REVIEW

File: onnxruntime/contrib_ops/cpu/transformers/beam_search_parameters.cc [hunk 1]
```diff
@@ -156,6 +190,8 @@ void WhisperBeamSearchParameters::ParseFromAttributes(const OpKernelInfo& info)
   no_speech_token = static_cast<int>(info.GetAttrOrDefault<int64_t>("no_speech_token", -1LL));
   cross_qk_layer_head_input_id = 12;
   extra_decoding_ids_input_id = 13;
+  left_pad_mask_input_id = 14;
```
Comment (shubhambhokare1) [inline onnxruntime/contrib_ops/cpu/transformers/beam_search_parameters.cc:193]: Change to 15/16 instead of input 14/15

File: onnxruntime/contrib_ops/cuda/transformers/beam_search.cc [hunk 2]
```diff
@@ -50,6 +50,8 @@ ONNX_OPERATOR_KERNEL_EX(
         .InputMemoryType(OrtMemTypeCPUInput, 10)   // 'decoder_input_ids' needs to be on CPU
         .InputMemoryType(OrtMemTypeCPUInput, 11)   // 'logits_processor' needs to be on CPU
         .InputMemoryType(OrtMemTypeCPUInput, 14)   // 'temperature' needs to be on CPU
+        .InputMemoryType(OrtMemTypeCPUInput, 15)   // 'left_pad_mask' needs to be on CPU
```
Comment (shubhambhokare1) [inline onnxruntime/contrib_ops/cuda/transformers/beam_search.cc:53]: Update new input descriptions

File: onnxruntime/python/tools/transformers/models/whisper/whisper_helper.py [hunk 3]
```diff
@@ -374,8 +381,15 @@ def verify_onnx(
             "tensor(uint8)": np.uint8,
         }
 
+        # Generate prompts
+        prompt_text = "Christians"
```
Comment (shubhambhokare1) [inline onnxruntime/python/tools/transformers/models/whisper/whisper_helper.py:385]: Clean up testing module

**Review by thiagocrepaldi — COMMENTED** [review]
Try running `lintrunner -a` to fix the lint issues.

File: onnxruntime/python/tools/transformers/onnx_model_bart.py [hunk 4]
```diff
@@ -124,7 +125,12 @@ class BartOnnxModel(BertOnnxModel):
     def __init__(self, model, num_heads, hidden_size, model_impl="hf"):
         super().__init__(model, num_heads, hidden_size)
         self.attention_mask = AttentionMask(self)
-        self.attention_fusion = FusionBartAttention(self, self.hidden_size, self.num_heads, self.attention_mask)
+        if model_impl == "openai":
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/onnx_model_bart.py:128]: I think we do not need model_impl. We can do like the following:
    fuse attention with FusionBartAttentionOpenai
    fuse attention with FusionBartAttention
    
So that it can handle graph patterns of both OpenAI and HF ones.

File: onnxruntime/python/tools/transformers/fusion_options.py [hunk 5]
```diff
@@ -59,6 +59,7 @@ def __init__(self, model_type):
 
         if model_type == "clip":
             self.enable_embed_layer_norm = False
+        self.model_impl = "hf"
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/fusion_options.py:62]: There is no need for this option. See another of my comment.

**Comment by thiagocrepaldi** [comment]
@shubhambhokare1 any update for this pr?

**Comment by shubhambhokare1** [comment]
https://github.com/microsoft/onnxruntime/pull/19854 has the same functionality without the requirement of adding new inputs to the subgraphs
